xcm/xcm-simulator/example/src/lib.rs

```diff
@@ -212,4 +212,62 @@ mod tests {
 			);
 		});
 	}
+
+	#[test]
+	fn recursion_limit_works() {
+		use parachain::{Event, System};
+		use xcm_executor::MAX_RECURSION_LIMIT;
+		MockNet::reset();
+
+		ParaA::execute_with(|| {
+			let mut msg = WithdrawAsset {
+				assets: vec![ConcreteFungible { id: X1(Parent), amount: 0 }],
+				effects: vec![],
+			};
+			for _ in 0..MAX_RECURSION_LIMIT {
```

Comment:
why are you assigning `msg` `MAX_RECURSION_LIMIT` times? this doesn't make sense to me.